chombot-common/src/tournaments_watcher/notifier.rs

```diff
@@ -13,16 +18,43 @@ pub trait TournamentsUpdateNotifier<R: Send + Sync> {
     async fn notify(&self, ranking: &R);
 }
 
-pub struct TournamentsChannelMessageNotifier {
-    channel_id: ChannelId,
+#[async_trait]
+pub trait TournamentWatcherChannelListProvider: Send + Sync {
+    type TournamentWatcherChannelList: IntoIterator<Item = ChannelId> + Send;
+
+    async fn tournament_watcher_channels(&self) -> Self::TournamentWatcherChannelList;
+}
+
+#[async_trait]
+impl TournamentWatcherChannelListProvider for ChannelId {
+    type TournamentWatcherChannelList = Once<Self>;
+
+    async fn tournament_watcher_channels(&self) -> Self::TournamentWatcherChannelList {
+        iter::once(*self)
+    }
+}
+
+#[async_trait]
+impl<T: TournamentWatcherChannelListProvider> TournamentWatcherChannelListProvider
+    for Arc<RwLock<T>>
+{
+    type TournamentWatcherChannelList = T::TournamentWatcherChannelList;
+
+    async fn tournament_watcher_channels(&self) -> Self::TournamentWatcherChannelList {
+        self.read().await.tournament_watcher_channels().await
+    }
+}
+
+pub struct TournamentsChannelMessageNotifier<T> {
+    channel_list_provider: T,
     message: String,
 }
 
-impl TournamentsChannelMessageNotifier {
+impl<T: TournamentWatcherChannelListProvider> TournamentsChannelMessageNotifier<T> {
     #[must_use]
-    pub const fn new(channel_id: ChannelId, message: String) -> Self {
+    pub const fn new(channel_list_provider: T, message: String) -> Self {
```

Comment:
To keep being consistent in our naming scheme:
```suggestion
    pub const fn new(tournament_watcher_channel_list_provider: T, message: String) -> Self {
```

||PLEASE DON'T ACCEPT IT||